app/models/comment.rb

```diff
@@ -65,19 +63,19 @@ def akismet_attributes
   after_create :update_work_stats
   after_destroy :update_work_stats
 
+  before_update :moderate_update
   after_update :after_update
+
+  # Put comments back into moderation after significant changes
+  def moderate_update
+    return if !moderated_commenting_enabled? || is_creator_comment?
+    self.unreviewed = true if significant_content_change?
+  end
+
   def after_update
     users = []
     admins = []
 
-    if self.saved_change_to_edited_at? && self.saved_change_to_comment_content? && self.moderated_commenting_enabled? && !self.is_creator_comment?
```

Comment:
This comment from the original PR still hasn't been addressed: https://github.com/otwcode/otwarchive/pull/3642#discussion_r329283994

Reply:
Thanks for fishing that out of the well! (I find it very hard to follow the many comments on giant PRs, GitHub should really come up with a better way to wade through them...)

I don't quite understand what is the desired outcome, though - the return of the old conditionals but in the new method, like so:
```
    def significant_content_change?
      saved_change_to_edited_at? &&
        saved_change_to_comment_content? &&
        content_too_different?(comment_content, comment_content_was)
    end
```
or adding back the `edited_at` check, but replacing `saved_change_to_comment_content?` with `comment_content_changed?`?

> what the issue was: `self.comment_content_was` vs `self.comment_content_before_last_save`

I don't understand what this means.

Reply:
I don't believe either of those fixes will work. Rails changed the behavior of `attribute_was` in 5.2, which is what [this deprecation warning that I copied from the 5.1 automated tests](https://github.com/otwcode/otwarchive/pull/3642#issuecomment-539723563) was about:
> DEPRECATION WARNING: The behavior of `attribute_was` inside of after callbacks will be changing in the next version of Rails. The new return value will reflect the behavior of calling the method after `save` returned (e.g. the opposite of what it returns now). To maintain the current behavior, use `attribute_before_last_save` instead. (called from after_update at /home/travis/build/tickinginstant/otwarchive/app/models/comment.rb:75)

The code from the original PR moves the check from `after_update` to `before_update`, which is why it uses `attribute_was` and `attribute_changed?` rather than `attribute_before_last_save` and `saved_change_to_attribute?`. But it sounds like you're proposing mixing them, which won't work regardless of when the check runs.

At this point, I think it'd be simplest to revert all of the changes to `comment.rb` (aside from the scope changes) and change the line `if content_too_different?(self.comment_content, self.comment_content_was)` to `if content_too_different?(comment_content, comment_content_before_last_save)`. That should also help prevent the Hound from barking about everything under the sun.

Reply:
Oh, I'm not proposing anything *g*, I was just trying to understand your original comment here https://github.com/otwcode/otwarchive/pull/3642#discussion_r329283994 and likely misinterpreting the `edited_at` bit.

(I didn't see the other comment because it wasn't directly on the code being discussed, and I never noticed the deprecation errors in the test suite. 😭 )

Reply:
Well, I've done that, and then the feature test was failing. I've added some `reload`s to the assertion and it passed locally. Who knows what Travis will make of it. 😩

Reply:
I think that's an intermittent error in the existing code -- I'm pretty sure I ran into the same error in a Travis run for a completely unrelated branch, though unfortunately I restarted the the job (rather than preserving the log) when I realized it was intermittent, so I can't verify that it was the right intermittent failure.

If I'm not mistaken, the intermittent error occurs because if the test runs too quickly, the second comment edit doesn't change `edited_at`, which means that `saved_change_to_edited_at?` doesn't return true. That means that the comment isn't treated like a normal edited comment is (and therefore doesn't have its status set back to `unreviewed`).

I added some debugging code to print out `edited_at` and `edited_at_before_last_save`, and ran the test on this branch several times. The success/failure was determined in each case by whether the two `edited_at` values matched. Unfortunately, I'm having trouble replicating the same behavior on the master branch -- the test on master seems to be running much slower, so I have yet to be able to replicate the intermittent failure. (Though I guess that's good news for the upgrade, if it runs faster?)

Reply:
I got this error pretty consistently if I remove the `reload`s from the unreviewed checking steps, was this the one you ran into?

https://travis-ci.com/github/redsummernight/otwarchive/jobs/379025244

Reply:
No, that's a different intermittent error that's been happening on a fairly regular basis. That's spam vs. not spam (the `approved` column), not reviewed vs. unreviewed (the `unreviewed` column). Not sure why the `reload` would affect that, since the steps aren't used there. Probably just luck.

The only tests that use the two `marked as unreviewed` steps are in `comment_moderation.feature`. The intermittent failure I was talking about was on this test, IIRC:
https://github.com/otwcode/otwarchive/blob/b850d356002a00c4b57a679f64baf03662bf629a/features/comments_and_kudos/comment_moderation.feature#L262-L283

The added `reload` definitely doesn't help resolve the the issue with that test, since the step already loads a fresh copy of the comment even without calling `reload`. It might help with the one test where the step is called twice, by acting as a delay to help ensure that the timestamps change, but if we want a delay it's probably better to explicitly wait 1 second.

Reply:
Gotcha, thanks! Took a few retries, but [this build](https://travis-ci.com/github/redsummernight/otwarchive/jobs/379025244) is passing now. I'll remove the `reload`s and we can add delays outside of this PR.